sharding/collation.go

```diff
@@ -22,17 +25,39 @@ type CollationHeader struct {
 	proposerSignature []byte          //the proposer's signature for calculating collation hash.
 }
 
+// Hash takes the keccak256 of the collation header's contents.
+func (h *CollationHeader) Hash() (hash common.Hash) {
+	hw := sha3.NewKeccak256()
+	rlp.Encode(hw, h)
+	hw.Sum(hash[:0])
+	return hash
+}
+
+// ShardID is the identifier for a shard.
+func (h *CollationHeader) ShardID() *big.Int { return h.shardID }
+
+// Period the collation corresponds to.
+func (h *CollationHeader) Period() *big.Int { return h.period }
+
+// ChunkRoot of the serialized collation body.
+func (h *CollationHeader) ChunkRoot() *common.Hash { return h.chunkRoot }
+
 // Header returns the collation's header.
 func (c *Collation) Header() *CollationHeader { return c.header }
 
-// Transactions returns an array of tx's in the collation.
-func (c *Collation) Transactions() []*types.Transaction { return c.transactions }
+// Body returns the collation's byte body.
+func (c *Collation) Body() []byte { return c.body }
 
-// ShardID is the identifier for a shard.
-func (c *Collation) ShardID() *big.Int { return c.header.shardID }
+// Hash returns the hash of a collation's entire contents. Useful for tests.
+func (c *Collation) Hash() (hash common.Hash) {
+	hw := sha3.NewKeccak256()
+	rlp.Encode(hw, c)
+	hw.Sum(hash[:0])
+	return hash
+}
 
-// Period the collation corresponds to.
-func (c *Collation) Period() *big.Int { return c.header.period }
+// Transactions returns an array of tx's in the collation.
+func (c *Collation) Transactions() []*types.Transaction { return c.transactions }
```

Comment:
We still need them because we'll be deserializing the chunks into transactions. We'll need a place to store this deserialized data and keeping a tx's slice is good here.